### PR TITLE
Allow creation and management of departments from the API

### PR DESCRIPTION
**Pull request by xsrust**
I've added 4 endpoints in v0 of the api to a new controller: `api/v0/departments_controller.rb`
- `index` displays an array of departments for any user's organization
- `create` allows an org-admin to specify a name and code to create a department
- `users` allows an org-admin to see which accounts (by email) are associated with each department.
- `assign_users` allows an org-admin to specify a list of users (`user[]` in params) to be assigned to a specific department (using the id returned by `index` and the email addresses from `users`)

### CODE REVIEW

File: app/controllers/api/v0/departments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      raise Pundit::NotAuthorizedError
+    end
+    @department = Department.new(org: @user.org,
+                                 code: params[:code],
+                                 name: params[:name])
+    if @department.save
+      redirect_to api_v0_departments_path
+    else
+      # the department did not save
+      self.headers["WWW-Authenticate"] = "Token realm=\"\""
```
Comment (briri) [inline app/controllers/api/v0/departments_controller.rb:20]: 👍 this great. The v1 json responses contain the error messages but we should probably also list them in the 'realm' as well.

File: app/controllers/api/v0/departments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @users = @user.org.users.includes(:department)
+  end
+
+  def assign_users
+    @department = Department.find(params[:id])
+
+    unless Api::V0::DepartmentsPolicy.new(@user, @department).assign_users?
+      raise Pundit::NotAuthorizedError
+    end
+
```
Comment (briri) [inline app/controllers/api/v0/departments_controller.rb:47]: Should probably do a nil check on `@department` here to make sure it doesn't accidentally blank out associations if the id was incorrect for some reason.

Comment (xsrust) [reply app/controllers/api/v0/departments_controller.rb:47]: Sure, I've added that on the policy check for assign_users?

**Comment by briri** [comment]
this looks fine @xsrust looks like it just needs a rebase. feel free to merge after that

**Pull request merged**